packages/libs/eda/src/lib/workspace/sharing/ShareFromAnalysis.tsx

```diff
@@ -45,6 +55,18 @@ export default function ShareFromAnalysis({
     sharingUrlPrefix
   ).href;
 
+  const sharingDatasetUrl = useMemo(() => {
+    if (!analysisState.analysis) return;
+    if (!isDiyWdkRecordId(analysisState.analysis.studyId)) return;
+    const datasetUrlHelper = window.location.pathname.split('/analyses')[0];
+    return new URL(
+      `${datasetUrlHelper}/datasets/${wdkRecordIdToDiyUserDatasetId(
+        analysisState.analysis.studyId
+      )}`,
+      sharingUrlPrefix
+    ).href;
+  }, [analysisState.analysis]);
+
```

Comment:
**question**: does `sharingDatasetUrl` need `useMemo`?

Reply:
Nope! I removed it. 🗑️